File: ParkHaeYoung/01월 2주차/택배상자.js
function solution(order) {
    let answer = 0;
    const stack = []; // 보조 컨테이너 역할

    for(let i = 1; i <= order.length; i++) {
        // 박스를 보조 컨테이너에 넣고 보기
        stack.push(i);

        // 보조 컨테이너의 마지막 값 === 현재 담아야 하는 박스
        while(stack.length > 0 && stack[stack.length - 1] === order[answer]) {
            stack.pop();
            answer++;
        }
    }
    return answer;
}

File: LeeNaYoung/1월 2주차/택배상자.py
def solution(order):
    answer = 0
    i = 0
    from collections import deque
    box = [0 for i in range(len(order))]
    for o in range(len(order)):
        box[order[o]-1] = o
    box=deque(box)
    truck = deque()
    trail = deque()
    
    while len(box) > 0:
        if box[0] == answer:
            truck.append(box.popleft())
            answer+=1
        elif len(trail)>0 and trail[-1] == answer:
            truck.append(trail.pop())
            answer+=1
        else:
            trail.append(box.popleft())
        if len(box)==0 and len(trail) > 0 and answer==trail[-1]:
            box.append(trail.pop())
        
    return answer
